src/Controls/src/Core/Element/Element.cs

```diff
@@ -627,13 +627,34 @@ protected virtual void OnParentSet()
 			(this as IPropertyPropagationController)?.PropagatePropertyChanged(null);
 		}
 
+		Dictionary<string,bool> _pendingHandlerUpdatesFromBPSet = new Dictionary<string, bool>();
+		private protected override void OnBindablePropertySet(BindableProperty property, object original, object value, bool changed)
+		{
+			_pendingHandlerUpdatesFromBPSet[property.PropertyName] = changed;
```

Comment:
What will happen here if there is a double set where the first is changed and the second is the same? Will that prevent any updates from firing? Should this rather be a set to true and then if false do nothing? Also maybe no use setting to false.

Is the value used at all? Maybe this is better in a hash list?

Reply:
AFAICT (and the tests I've added show it) the BP gates multiple setters. So, the second setvalue isn't going to start until the first one finishes. Ideally, we could debounce in this scenario and only send the second update, but, let's do that in a followup PR